src/modules/app/actions/init-augur.js

```diff
@@ -31,44 +34,71 @@ function pollForAccount(dispatch) {
   }, 250)
 }
 
-function pollForNetwork(dispatch, expectedNetwork) {
+function pollForNetwork(dispatch, getState) {
   let networkId
 
   setInterval(() => {
+    const { modal, env } = getState()
+    const expectedNetwork = env['network-id']
     if (parseInt(AugurJS.augur.rpc.getNetworkID(), 10) !== networkId) {
       networkId = parseInt(AugurJS.augur.rpc.getNetworkID(), 10)
 
-      if (networkId !== expectedNetwork) {
+      if (networkId !== expectedNetwork && !!modal.type && modal.type !== MODAL_NETWORK_DISCONNECTED) {
         dispatch(updateModal({
           type: MODAL_NETWORK_MISMATCH,
           expectedNetwork
         }))
-      } else {
+      } else if (!!modal.type && modal.type === MODAL_NETWORK_MISMATCH) {
         dispatch(closeModal())
       }
     }
   })
 }
 
-export function initAugur(callback = logError) {
+export function reconnectAugur(history, env, callback = logError) {
+  return (dispatch, getState) => {
+    AugurJS.connect(env, (err, ConnectionInfo) => {
+      if (err || !ConnectionInfo.augurNode || !ConnectionInfo.ethereumNode) {
+        return callback(err, ConnectionInfo)
+      }
+      const ethereumNodeConnectionInfo = ConnectionInfo.ethereumNode
+      dispatch(updateConnectionStatus(true))
+      dispatch(updateContractAddresses(ethereumNodeConnectionInfo.contracts))
+      dispatch(updateFunctionsAPI(ethereumNodeConnectionInfo.abi.functions))
+      dispatch(updateEventsAPI(ethereumNodeConnectionInfo.abi.events))
+      dispatch(updateAugurNodeConnectionStatus(true))
+      dispatch(registerTransactionRelay())
+      dispatch(loadUniverse(env.universe || AugurJS.augur.contracts.addresses[AugurJS.augur.rpc.getNetworkID()].Universe, history))
+      // close any modals that may be open because of reconnection
+      dispatch(closeModal())
+      // shouldn't need these as they should be running already...
+      // pollForAccount(dispatch, getState)
+      // pollForNetwork(dispatch, getState, env['network-id'])
+      callback()
+    })
+  }
+}
+
+export function initAugur(history, callback = logError) {
   return (dispatch, getState) => {
     const xhttp = new XMLHttpRequest()
     xhttp.onreadystatechange = () => {
       if (xhttp.readyState === 4) {
         if (xhttp.status === 200) {
           const env = JSON.parse(xhttp.responseText)
           dispatch(updateEnv(env))
-          AugurJS.connect(env, (err, ethereumNodeConnectionInfo) => {
+          AugurJS.connect(env, (err, ConnectionInfo) => {
```

Comment:
A couple slight structural changes here —

* Rename `reconnectAugur` -> `connectAugur`
* Dispatch `connectAugur` from w/in `initAugur`, effectively DRYing things up.

Reply:
👍